### DISCUSSION TITLE
LEDs stay on all the time, even if the tv is off

### DISCUSSION THREAD

**Discussion by grannymaster** [awawa-dev/HyperHDR#304] [Q&A] [answered]
Hey, 

I have a problem where the LEDs stay on even if the TV is turned off.

My HyperHDR info: https://i.imgur.com/DNR8Yap.png
My Capture settings: https://i.imgur.com/xFocGdh.png
My general setup: https://i.imgur.com/OPkPkEK.png

If I turn off CEC and then back on, the LEDs will turn off for a moment, but then light up again.

I'm using this LED strip: https://www.amazon.de/-/en/gp/product/B078S6Z9KG/ref=ewc_pr_img_3?smid=A2CRFKF5XF9GCG&psc=1

Is there an option I missed or is this a bug?

**Comment by awawa-dev** [comment]
Hi
Need FULL HyperHDR logs with a captured moment when the TV goes off (that triggers CEC OFF response in HyperHDR)

**Reply by grannymaster** [reply]
Here is the full log, I cycled the setup from off->on-off

https://pastebin.com/zr67GXRU

**Reply by grannymaster** [reply]
The timestamps are wrong for some reason.

**Comment by awawa-dev** [comment]
Also try to update to v18 beta3 first. Some things related to turning off LED devices were fixed. Maybe it will help in your case too.

**Reply by grannymaster** [reply]
NVM, figured it out.

Same thing happening, here is the log:
https://pastebin.com/BKMHS5t7

**Comment by awawa-dev** [comment]
OK, from your logs your CEC support in your system is completely dead. Probably the only reaction from the LED strip when you turn off the TV is from the USB grabber which detect TV disconnection and hangs for a moment, not from CEC library. There is a lot of CEC errors here and unfortunately it seems the external libCEC is in half-abandon state. HyperHDR ships its own copy with some patches provided by the community. Alternatively you can try to switch for OS CEC:
1 install ```sudo apt install libcec6```
2 if it succeeded then remove files in \usr\share\hyperhdr\lib folder: libcec.so.6.0.2 libcec.so.6 libcec.so

BTW another thing from the log: your auto-detection of the no-signal needs re-calibration. You need to recalibrate it every time you change the capturing stream resolution for example switch from MJPEG to YUv or change the resolution. So in this case nor CEC nor the auto-detection signal don't work properly.

```2022-07-22T16:25:32.566Z [SIGNAL_AUTO] Calibration data is not meant for the current video stream (different dimensions). Please run the calibration procedure in the panel grabber tab.```

**Reply by grannymaster** [reply]
The CEC library didn't help, but I found a fix. I just automated it with Home Assistant, works like a charm.

Another quick question, my dad wants to start user HyperHDR, but he doesn't have HOAS to fix the CEC. Is there a post or something which has all the parts linked which would perfectly? Including CEC, HDR, Atmos etc.

**Reply by awawa-dev** [reply]
Normally CEC works fine with Rpi. There are some problems with particular TV models but in your case the CEC library refuses to initialize. Maybe you can give a try aarch64 version if you have a spare SD card? 

Properly configured signal detection can be a substitute for CEC also.

**Comment by awawa-dev** [comment, marked as answer]
I completely missed that in your scheme but to have CEC working you have to connect RPI hdmi to the TV directly. CEC works over HDMI only.

**Reply by grannymaster** [reply]
Haha :D How did I miss that.

Plugged an HDMI cable, now the LEDs dim, but don't turn off completely.

**Reply by awawa-dev** [reply]
need logs

**Reply by grannymaster** [reply]
Nevermind, figured it out.

It was Samsung TV OS being stupid as alwyas... Everything works now!
